consensus/src/block_storage/execution_pool_test.rs

```diff
@@ -0,0 +1,734 @@
+
+    //       ╭--> A1--> A2--> A3
+    // Genesis--> B1--> B2
+    //             ╰--> C1
+    let (_, block_store, pipelined_blocks) = create_block_tree_with_forks(window_size).await;
+    let [_, a1, a2, a3, b1, _, c1] = pipelined_blocks;
+
+    let ordered_root_pipelined_block = block_store.ordered_root();
+    assert_eq!(ordered_root_pipelined_block.round(), 0);
+
+    //             ┌──────────┐
+    // Genesis ──> │ A1 -> A2 │ ──> A3
+    //             └──────────┘
+    let ordered_blocks =
+        get_blocks_from_block_store_and_window(block_store.clone(), a3.block(), window_size);
+    assert_eq!(ordered_blocks.len(), 2);
+    assert_eq!(ordered_blocks.first().unwrap().id(), a1.id());
+    assert_eq!(ordered_blocks.get(1).unwrap().id(), a2.id());
+
+    //             ┌────┐
+    // Genesis ──> │ B1 │ ──> C1
+    //             └────┘
+    let ordered_blocks =
+        get_blocks_from_block_store_and_window(block_store.clone(), c1.block(), window_size);
+    assert_eq!(ordered_blocks.len(), 1);
+    assert_eq!(ordered_blocks.first().unwrap().id(), b1.id());
+}
+
+#[tokio::test]
+async fn test_execution_pool_window_size_greater_than_block_store() {
+    // window size > block store size
+    const NUM_BLOCKS: usize = 4;
+    let window_size: usize = 10;
+
+    // Genesis ──> A1 ──> A2 ──> A3
+    let (_, block_store, pipelined_blocks) =
+        create_block_tree_no_forks::<{ NUM_BLOCKS }>(NUM_BLOCKS as u64, window_size).await;
+    let [_, a1, a2, a3] = pipelined_blocks;
+
+    //            ┌───────────┐
+    // Genesis ─> │ A1 ──> A2 │ ──> A3
+    //            └───────────┘
+    let blocks =
+        get_blocks_from_block_store_and_window(block_store.clone(), a3.block(), window_size);
+
+    assert_eq!(blocks.len(), 2);
+    assert_eq!(blocks.first().unwrap().id(), a1.id());
+    assert_eq!(blocks.get(1).unwrap().id(), a2.id());
+}
+
+#[tokio::test]
+async fn test_execution_pool_block_window_with_pruning() {
+    const NUM_BLOCKS: usize = 5;
+    let window_size: usize = 3;
+
+    // Genesis ──> A1 ──> ... ──> A4
+    let (_, block_store, pipelined_blocks) =
+        create_block_tree_no_forks::<{ NUM_BLOCKS }>(NUM_BLOCKS as u64, window_size).await;
+    let [_, _, a2, a3, a4] = pipelined_blocks;
+
+    // A2 ──> ... ──> A4
+    block_store.prune_tree(a2.id());
+    let ordered_root = block_store.ordered_root();
+    let commit_root = block_store.commit_root();
+    assert_eq!(ordered_root.round(), 2);
+    assert_eq!(commit_root.round(), 2);
+
+    // ┌───────────┐
+    // │ A2 ──> A3 │ ──> A4
+    // └───────────┘
+    let blocks =
+        get_blocks_from_block_store_and_window(block_store.clone(), a4.block(), window_size);
+    assert_eq!(blocks.len(), 2);
+    assert_eq!(blocks.first().unwrap().id(), a2.id());
+    assert_eq!(blocks.get(1).unwrap().id(), a3.id())
+}
+
+/// `get_block_window` on a block that has been pruned. Should panic.
+/// TODO revisit this test. A pruned node should no longer exist in the
```

Comment:
This might be because of `max_pruned_blocks_in_mem` some of the pruned blocks are still kept, we could set this to 0 for the test

Reply:
Done